core/src/server/helpers.rs

```diff
@@ -226,6 +233,16 @@ impl MethodResponseResult {
 }
 
 impl MethodResponse {
+	/// Create a subscription response.
+	pub fn subscription_response<T>(id: Id, result: ResponsePayload<T>, max_response_size: usize) -> Self
+	where
+		T: Serialize + Clone,
+	{
+		let mut rp = Self::response(id, result, max_response_size);
+		rp.is_subscription = true;
+		rp
+	}
+
 	/// Send a JSON-RPC response to the client. If the serialization of `result` exceeds `max_response_size`,
 	/// an error will be sent instead.
 	pub fn response<T>(id: Id, result: ResponsePayload<T>, max_response_size: usize) -> Self
```

Comment:
Nit: I wonder whether it's worth renaming this to `method_response` to try and make it feel like "not the default one to use", and/or maybe just a tweak to docs like "If the response is from a subscription, use `subscription_response`", orr ditching `subscription_response` and instead adding another arg to `response` (ie `is_subscription: bool`) so that every call has to be explicit.

Reply:
Sure, it makes sense. 

I just added `subscription_response` and never thought about it. I was just happy that it worked :P

Reply:
I'm not too fussed, having looked at the rest of the code (maybe just add the doc comment?), so whatever you prefer :)

Reply:
yeah, a subscription response is basically just a method response but in some scenarios in the server it may be useful to know and in middleware doing grafana metrics etc.

I think I want to keep response but lemme adjust the docs a little bit better.